src/shared/components/tree/DescriptorTreeNode.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import * as styles from './styles.module.scss';
+import classNames from '../../lib/classNames';
+import FontAwesome from "react-fontawesome";
+
+export interface ITreeDescriptor<TreeNode>
+{
+    isExpanded(node:TreeNode):boolean;
+    getContent(node:TreeNode):React.ReactChild;
+    getChildren(node:TreeNode):TreeNode[]|undefined;
+}
+
+export interface IDescriptorTreeNodeProps<TreeNode>
+{
+    treeDescriptor: ITreeDescriptor<TreeNode>;
+    tree: TreeNode;
+
+    className?: string;
+    style?: React.CSSProperties;
+    onExpand?: (node:TreeNode, expand:boolean)=>void;
+    isChild?: boolean;
+}
+
+export interface IDescriptorTreeNodeState
+{
+	expand?: boolean;
+}
+
+export default class DescriptorTreeNode extends React.Component<IDescriptorTreeNodeProps<any>, IDescriptorTreeNodeState>
+{
+	constructor(props:IDescriptorTreeNodeProps<any>)
+	{
+		super(props);
+		this.state = {};
+	}
+
+	onExpand = () =>

```

Comment:
actually, it looks like this isn't used, right?  onExpand seems to come from props